pymc/distributions/distribution.py

```diff
@@ -371,7 +370,7 @@ def get_moment(rv: TensorVariable) -> TensorVariable:
     for which the value is to be derived.
     """
     size = rv.owner.inputs[1]
-    return _get_moment(rv.owner.op, rv, size, *rv.owner.inputs[3:])
+    return _get_moment(rv.owner.op, rv, size, *rv.owner.inputs[3:]).astype(aesara.config.floatX)
```

Comment:
Good placement!

Reply:
Actually just realized it should convert to the RVs dtype which maybe be int64